File: joseffe-barroso/Python/aula04/while/exercise42.py
sequence = int(input("Qual é o valor do termo?"))

while (sequence <= 0 or sequence > 50):
    sequence = int(input("O valor da sequência de termos deve ser positivo e menor que cinquenta. Informe novamente: "))

numerator = 1
denominator = 2
index = 1
sumNumbers = 0

while (index < sequence):
    division = numerator / denominator
    sumNumbers = sumNumbers + division
    print(f"{index}. {numerator}/{denominator}")
    numerator = numerator + 1
    denominator = denominator + 1
    index = index + 1

print("A soma deu: %.2f" % sumNumbers)

File: joseffe-barroso/Java/src/aula04/While/Exercise42.java
package While;

import java.util.Scanner;

public class Exercise42 {

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		int n, i;
		double dividendo, divisor, divisao, soma;
		
		System.out.println("Digite o termo que quer encontrar (valor deve ser positivo e menor que 50): ");
		n = scanner.nextInt();
		
		while ((n <= 0) || (n > 50)) {
			System.out.println("O valor de N deve ser positivo e menor que 50.");
			System.out.println("Digite o termo que quer encontrar (valor deve ser positivo e menor que 50): ");
			n = scanner.nextInt();
		}
		
		dividendo = 1;
		divisor = 2;
		i = 1;
		soma = 0;
		
		while (i <= n) {
		    divisao = dividendo / divisor;
		    soma = soma + divisao;
    	    System.out.println(i + ". " + (int) dividendo + "/" + (int) divisor);
    	    dividendo++;
    	    divisor++;
    	    i++;
		}
		System.out.printf("A soma é: %.2f", soma);
		
		scanner.close();
	}

}
